### PR TITLE
Enable Arcs Cache Manager on DevEx

### PR DESCRIPTION
**Pull request by ianull**
Check go/arcs-cache-manager for optimization results.
After the ServiceWorker officially supports dedicated workers (soon), the latency will be improved
significantly (10-50+ ms) for particles' execution of each PEC spawned at a new dedicated worker context at run-time. 

Turn on for the time being and follow up bugs if any.

### CODE REVIEW

File: javaharness/java/arcs/android/AndroidRuntimeSettings.java [hunk 1]
```diff
@@ -32,13 +32,18 @@
...
+  private static final String DEFAULT_SHELL_URL =
+      "https://appassets.androidplatform.net/assets/arcs/index.html?";
   // Load the on-device assets
   private static final boolean DEFAULT_ASSETS_FROM_WORKSTATION = false;
   // Uses the standard 8786 port
   private static final int DEFAULT_DEV_SERVER_PORT = 8786;
-  // Deactivates the Arcs Cache Manager.
-  private static final boolean DEFAULT_USE_CACHE_MANAGER = false;
+  // Activates the Arcs Cache Manager.
+  private static final boolean DEFAULT_USE_CACHE_MANAGER = true;
```
Comment (shaper) [inline javaharness/java/arcs/android/AndroidRuntimeSettings.java]: If we need to turn this off again for some reason, will a subsequent PR to do that just need to flip this to `false`, or do we need to do some of the other edits in this PR (such as the one in `cache-mgr-sw.js` below)?

Comment (ianull) [reply javaharness/java/arcs/android/AndroidRuntimeSettings.java]: To turn off this feature temporarily, using the single adb command:
> adb shell "setprop debug.arcs.runtime.use_cache_mgr false"
The hardcoded DEFAULT_USE_CACHE_MANAGER will be overridden by this command.

To rollback to file:/// protocol, using the single adb command:
> adb shell "setprop debug.arcs.runtime.shell_url 'file:///android_asset/arcs/index.html?'"

Combine these two commands will roll everything dynamically back to that before the change.
It's convenient for devs to verify w/ vs. w/o the feature.

**Comment by ianull** [comment]
@shaper sure, will PSA in chat.

File: javaharness/java/arcs/android/AndroidRuntimeSettings.java (hunk 1 above)
Comment (ianull) [reply javaharness/java/arcs/android/AndroidRuntimeSettings.java]: To turn off this feature for good, yes, flip the flag to false in new PR is good enough (no service worker is created at run-time anymore).

**Pull request merged**